### DISCUSSION TITLE
Why don't `timed_mutex`, `recursive_timed_mutex`, and `shared_timed_mutex` have `native_handle`?

### DISCUSSION THREAD

**Discussion by frederick-vs-ja** [microsoft/STL#3387] [General] [answered]
Members `native_handle` and `native_handle_type` are optional for some classes in the concurrency support library. Currently MSVC STL provides them for some of these classes, but not all. Is this an intentional design?

**Comment by CaseyCarter** [comment, marked as answer]
Yes, I believe this is intentional. The three subject types are all implemented with a `mutex`, a `condition_variable`, and some other small bits of state; there's nothing that `native_handle` could usefully return. It's arguably bogus that `mutex`, `recursive_mutex`, and `condition_variable` return a `native_handle` that's only usable by going out of contract and calling library internals, I think anything we could do for the `meow_timed_mutex`es would be even worse.

To my knowledge, the intent of having `native_handle` for the types in the threading library is that one can get at the underlying POSIX thread library objects (hopefully) used to implement the C++ types so C code can interact with the same synchronization primitives as C++ code. This isn't the case for our platform, so there isn't much we can usefully do.
